core/blockchain.go

```diff
@@ -2569,6 +2580,29 @@ func (bc *BlockChain) trustedDiffLayerLoop() {
 	}
 }
 
+func (bc *BlockChain) startDoubleSignMonitor() {
+	eventChan := make(chan ChainHeadEvent, monitor.MaxCacheHeader)
+	sub := bc.SubscribeChainHeadEvent(eventChan)
+	headerChan := make(chan *types.Header, monitor.MaxCacheHeader)
+	defer func() {
+		sub.Unsubscribe()
+		bc.doubleSignMonitor.Close()
+		close(eventChan)
+		close(headerChan)
+		bc.wg.Done()
+	}()
+
+	go bc.doubleSignMonitor.Start(headerChan)
```

Comment:
why not just pass `eventChan` into `bc.doubleSignMonitor.Start` so we dont have to write following for loop.

Reply:
fixed